Sources/Bot/Services/DispatchService.swift

```diff
@@ -80,16 +80,22 @@ public final class DispatchService {
 
     private func pullRequestDidChange(event: PullRequestEvent) {
         logger.log("📣 Pull Request did change \(event.pullRequestMetadata) with action `\(event.action)`")
+
         let targetBranch = event.pullRequestMetadata.reference.target.ref
         let existingService = mergeServices.modify { (dict: inout [String: MergeService]) -> MergeService? in
             if let service = dict[targetBranch] {
                 // If service was already existing, return it so we'll send the pullRequestChangesObserver event outside this `modify` below
                 return service
             } else {
+                // Only create a new MergeService if the PullRequest for which we received an event is supposed to be included
+                guard case .include(let pullRequest)? = MergeService.Event.Outcome(event: event, integrationLabel: self.integrationLabel) else {
```

Comment:
Aren't we leaking implementation details in doing this filtering at this point? Because `MergeService.Event.Outcome` should be something private to `MergeService` and not exposed even internally, same for Events which are now exposed. 

I understand that we may end up creating a new `MergeService` without a need for it, if the PR is not labelled with the integration label, but that's an implementation detail and the `MergeService` will be released eventually so that's not entirely harmful in creating this instance.

Reply:
So you're suggesting we create the MergeService every time a new PR targeting a new branch is created, even if that merge service will directly go from `.starting` to `.idle` and be deleted right away?

I'm ok with that, just wondering if it's worth it compared to refactor that to just expose a `MergeService.shouldInclude(…) -> Bool` func (that would privately check if the `Outcome` is `.included` but will keep `Outcome` private?

Reply:
Either we do it for all events or none in my opinion, as a reader/maintainer is very confusing why we only filter part of the events... Because if the service already exists we just forward the change to the service which will eventually filter the event internally in the feedback that I've mentioned.

```swift
existingService?.pullRequestChangesObserver.send(value: (event.pullRequestMetadata, event.action))
```

Conceptually `DispatchService` should only be responsible by keeping the merge services and forward the events accordingly, now it's already doing some logic where it becomes aware of the concept of filtering Pull Requests that are not labelled with a certain integration label which is a concept that's expected to be part of the service itself.

Reply:
Yes I agree that I'm not a fan of that logic of only forwarding the event to the service if it already existed and use the `initialPullRequests` otherwise. We might want to improve that indeed.

I'll try to work on this, allowing the `.starting` state to also be able to listen and react to `pullRequestEvents`. _(Tried that in the original implementation but let to some unexpected complications about tracking state changes during startup sequence 😉 will try to revisit now that the code has settled)_

Reply:
> I'll try to work on this, allowing the .starting state to also be able to listen and react to pullRequestEvents.

I'm not entirely following, we are injecting `initialPullRequests` to the `MergeService` as part of the initialisation so that filtering can be done directly in the initialiser or inside the feedback for `starting` status, I'm not sure why we need to listen and react to `pullRequestEvents` 😕

Reply:
~We can't really do the filtering about the type of PullRequest event in `initialPullRequests` since we'd have the PullRequests directly here, not the event, so we wouldn't be able to check if the event leading to this was a `.opened` or `.labeled` or whatnot~

Wait or were you suggesting to just filter on the `pullRequest.isLabelled(state.integrationlabel)` – instead of filtering on the event? (I thought that at first but was afraid of possible edge cases, but in fact maybe that could be enough indeed 🤔 )

Reply:
Events are irrelevant for the initial list of pull requests, the only thing that matters is if the PR has the integration label or not, during the start up of the bot we don't have any events being fired, we just load the entire list of pull requests and start from there based on the presence of the integration label.

Reply:
Gotcha. But that means if we go that route, we'll keep the `existingService?.pullRequestChangesObserver.send` iow the fact that the `DispatchService` knows when to send those events (when MergeService was already existing) vs when not to (when it just got created), and I didn't like that either and wanted to improve this. But maybe I can send the event in all cases and filter in the `Feedback` if it's `.starting` after all, so this should work.